File: Reverse pattern printing.py
"""
input:
5
output:
15 10 6 3 1
14 9 5 2
13 8 4
12 7
11
"""
n=int(input())
first=int((n*(n+1))/2)
for i in range(0,n):
    val=n
    temp=first
    print(temp,end=" ")
    for j in range(0,n-i-1):
        print(temp-val,end=" ")
        temp=temp-val
        val-=1
    first-=1
    print()

File: Reverse pattern printing.cpp
#include <bits/stdc++.h>

using namespace std;

int main(int argc, char** argv)
{
int n;
cin>>n;
int first=(n*(n+1))/2;
for(int i=0;i<n;i++)
{
    int val=n;
    int temp=first;
    cout<<temp<<" ";
    for(int j=0;j<n-i-1;j++)
    {
        cout<<temp-val<<" ";
        temp=temp-val;
        val-=1;
    }
    first-=1;
    cout<<endl;
}

}
